src/Microsoft.Health.Fhir.TemplateManagement/Overlay/OverlayOperator.cs

```diff
@@ -0,0 +1,181 @@
+    {
+        public OCIFileLayer ExtractOCIFileLayer(OCIArtifactLayer artifactLayer)
+        {
+            EnsureArg.IsNotNull(artifactLayer, nameof(artifactLayer));
+
+            if (artifactLayer.Content == null)
+            {
+                return new OCIFileLayer();
+            }
+
+            var artifacts = StreamUtility.DecompressTarGzStream(new MemoryStream(artifactLayer.Content));
+            var metaBytes = artifacts.GetValueOrDefault(Constants.OverlayMetadataFile);
+            OverlayMetadata metaJson = null;
+            if (metaBytes != null)
+            {
+                try
+                {
+                    metaJson = JsonConvert.DeserializeObject<OverlayMetadata>(Encoding.UTF8.GetString(metaBytes));
+                }
+                catch (Exception ex)
+                {
+                    throw new OverlayException(TemplateManagementErrorCode.OverlayMetaJsonNotFound, $"{Constants.OverlayMetadataFile} file invalid", ex);
+                }
+            }
+
+            artifacts.Remove(Constants.OverlayMetadataFile);
+            return new OCIFileLayer() { Content = artifactLayer.Content, FileContent = artifacts, FileName = artifactLayer.FileName, SequenceNumber = metaJson == null ? -1 : metaJson.SequenceNumber, Digest = artifactLayer.Digest, Size = artifactLayer.Size };
+        }
+
+        public List<OCIFileLayer> ExtractOCIFileLayers(List<OCIArtifactLayer> artifactLayers)
+        {
+            EnsureArg.IsNotNull(artifactLayers, nameof(artifactLayers));
+
+            return artifactLayers.Select(ExtractOCIFileLayer).ToList();
+        }
+
+        public List<OCIFileLayer> SortOCIFileLayersBySequenceNumber(List<OCIFileLayer> fileLayers)
+        {
+            EnsureArg.IsNotNull(fileLayers, nameof(fileLayers));
+
+            return fileLayers.OrderBy(layer => layer.SequenceNumber <= -1 ? int.MaxValue : layer.SequenceNumber).ToList();
+        }
+
+        public OCIFileLayer MergeOCIFileLayers(List<OCIFileLayer> sortedLayers)
+        {
+            EnsureArg.IsNotNull(sortedLayers, nameof(sortedLayers));
+
+            List<string> removedFiles = new List<string>();
+            Dictionary<string, byte[]> mergedFiles = new Dictionary<string, byte[]> { };
+
+            for (var sequenceNumber = sortedLayers.Count - 1; sequenceNumber >= 0; sequenceNumber--)
+            {
+                var oneLayer = sortedLayers[sequenceNumber];
+                var fileContent = oneLayer.FileContent;
+                foreach (var oneFile in fileContent)
+                {
+                    if (oneFile.Value == null)
+                    {
+                        removedFiles.Add(oneFile.Key);
+                    }
+                    else if (!removedFiles.Contains(oneFile.Key) && !mergedFiles.ContainsKey(oneFile.Key))
+                    {
+                        mergedFiles.Add(oneFile.Key, oneFile.Value);
+                    }
+                }
```

Comment:
i feel you can add a function in OCI FileLayer, e.g. AddFile(name,content,overwrite)